### ISSUE TITLE
Cluster creation fails

### ISSUE THREAD

**Issue by satnam6502** [GoogleCloudPlatform/kubernetes#7537]
After making a recent build I found that I could not create a cluster.

```
$ kubectl version
Client Version: version.Info{Major:"0", Minor:"15", GitVersion:"v0.15.0", GitCommit:"831f3e60d7cd64c61a775d6c78acce1673dd8aa9", GitTreeState:"clean"}
Server Version: version.Info{Major:"0", Minor:"16+", GitVersion:"v0.16.0-46-gee27094238dbd7-dirty", GitCommit:"ee27094238dbd797c363bac6a54bc28a2cf71225", GitTreeState:"dirty"}

...
Kubernetes cluster is running.  The master is running at:

  https://104.197.49.90

The user name and password to use is located in /usr/local/google/home/satnam/.config/gcloud/kubernetes/kubeconfig.

... calling validate-cluster
Starting cluster using os distro: debian
Starting cluster using os distro: debian
current-context: "kubernetes-satnam_kubernetes"
Running: cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../_output/dockerized/bin/linux/amd64/kubectl get nodes -o template -t {{range.items}}{{.metadata.name}}
{{end}} --api-version=v1beta3
Found 4 nodes.
     1	kubernetes-minion-hhbv
     2	kubernetes-minion-jy35
     3	kubernetes-minion-lfcu
     4	kubernetes-minion-r2ez
Starting cluster using os distro: debian
current-context: "kubernetes-satnam_kubernetes"
Running: cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../_output/dockerized/bin/linux/amd64/kubectl get cs
Cluster not working yet.
Starting cluster using os distro: debian
current-context: "kubernetes-satnam_kubernetes"
Running: cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../_output/dockerized/bin/linux/amd64/kubectl get cs
Cluster not working yet.
Starting cluster using os distro: debian
current-context: "kubernetes-satnam_kubernetes"
Running: cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../_output/dockerized/bin/linux/amd64/kubectl get cs
Cluster not working yet.
Starting cluster using os distro: debian
current-context: "kubernetes-satnam_kubernetes"
Running: cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../_output/dockerized/bin/linux/amd64/kubectl get cs
Cluster not working yet.
Starting cluster using os distro: debian
current-context: "kubernetes-satnam_kubernetes"
Running: cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../_output/dockerized/bin/linux/amd64/kubectl get cs
Cluster not working yet.
Starting cluster using os distro: debian
current-context: "kubernetes-satnam_kubernetes"
Running: cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../_output/dockerized/bin/linux/amd64/kubectl get cs
 Validate output:
NAME                 STATUS      MESSAGE   ERROR
node-0               Healthy     ok        nil
node-1               Healthy     ok        nil
node-2               Unhealthy             Get https://kubernetes-minion-lfcu:10250/healthz: dial tcp 10.240.7.38:10250: connection refused
node-3               Healthy     ok        nil
controller-manager   Healthy     ok        nil
scheduler            Healthy     ok        nil
etcd-0               Healthy     {"action":"get","node":{"dir":true,"nodes":[{"key":"/registry","dir":true,"modifiedIndex":3,"createdIndex":3}]}}
                     nil
Validation returned one or more failed components. Cluster is probably broken.
```

**Comment by satnam6502** [comment]
I tore this cluster down and rebuilt it -- this time it worked. However, it is worrying that an attempt failed. We should try harder to make sure we can always create a working cluster.

**Comment by quinton-hoole** [comment]
Looks like a duplicate of #7302 ?

**Comment by zmerlynn** [comment]
As I mentioned in https://github.com/GoogleCloudPlatform/kubernetes/issues/7302#issuecomment-96731677, we do actually repeat health checks. You can see it clearly demonstrated here, we just don't spit out the `Validate output` each time (only on the final failure, so it's not entirely spammy). Closing #7302 as invalid.

**Comment by zmerlynn** [comment]
(cc @fabioy)

**Comment by zmerlynn** [comment]
This bug, unfortunately, is fairly unactionable. The output told you exactly which node to poke at, and pretty much how borked it was (we couldn't dial `/healthz`, so kubelet was totally unreachable). The cluster is already down. We should look at further first failure data capture we can go in this instance, but we need to keep it provider agnostic, or at least outside the shell scripts, so that it can work on GKE as well.

**Comment by quinton-hoole** [comment]
Looks like this is potentially fixed by  #7460 ? Reopen if not.
